pkg/server/server.go

```diff
@@ -51,9 +53,10 @@ type getEventsListener struct {
 	events chan *tetragon.GetEventsResponse
 }
 
-func NewServer(ctx context.Context, notifier notifier, observer observer) *Server {
+func NewServer(ctx context.Context, wg *sync.WaitGroup, notifier notifier, observer observer) *Server {
 	return &Server{
 		ctx:      ctx,
+		wg:       wg,
```

Comment:
How about renaming `wg` to `ctxCleanupWG` or something similar to make it clear that it's the waitgroup for performing any cleanup necessary if `ctx` is completed (assuming my understanding is correct).